submissions/ses3332002/friends-app/app.js

```diff
@@ -48,28 +57,39 @@ function changeData({target}) {
   if (target.tagName != "INPUT") {
     return;
   };
-  if (target.id == "as_is") {
-    sortCriterion = () => true;
-  } else if (target.id == "nameAZ") {
-    sortCriterion = (itemA, itemB) => itemA.name.first > itemB.name.first ? 1 : -1;
-  } else if (target.id == "nameZA") {
-    sortCriterion = (itemA, itemB) => itemA.name.first < itemB.name.first ? 1 : -1;
-  } else if (target.id == "age19") {
-    sortCriterion = (itemA, itemB) => itemA.dob.age > itemB.dob.age ? 1 : -1;
-  } else if (target.id == "age91") {
-    sortCriterion = (itemA, itemB) => itemA.dob.age < itemB.dob.age ? 1 : -1;
-  } else if (target.id == "all") {
-    filterCriterionSex = () => true;
-  } else if (target.id == "man") {
-    filterCriterionSex = (item) => item.gender == "male";
-  } else if (target.id == "woman") {
-    filterCriterionSex = (item) => item.gender == "female";
-  } else if ((target.id == "age_from")||(target.id == "age_to")) {
-    if ((age_from.value)&&(age_to.value)) {
-      filterCriterionAge = (item) => (item.dob.age >= age_from.value)&&(item.dob.age <= age_to.value);
-    };
-  } else if (target.id == "search_name") {
-    filterCriterionName = (item) => (item.name.first.toLowerCase().includes(search_name.value.toLowerCase()))||(item.name.last.toLowerCase().includes(search_name.value.toLowerCase()));
+  switch (target.id) {
+    case "as_is":
+      sortCriterion = () => false;
+      break;
+    case "nameAZ":
+      sortCriterion = (itemA, itemB) => itemA.name.first > itemB.name.first ? 1 : -1;
+      break;
+    case "nameZA":
+      sortCriterion = (itemA, itemB) => itemA.name.first < itemB.name.first ? 1 : -1;
+      break;
+    case "age19":
+      sortCriterion = (itemA, itemB) => itemA.dob.age > itemB.dob.age ? 1 : -1;
+      break;
+    case "age91":
+      sortCriterion = (itemA, itemB) => itemA.dob.age < itemB.dob.age ? 1 : -1;
+      break;
+    case "all":
+      filterCriterionSex = () => true;
+      break;
+    case "man":
+      filterCriterionSex = (item) => item.gender == "male";
+      break;
+    case "woman":
+      filterCriterionSex = (item) => item.gender == "female";
+      break;
+    case "age_from":
+    case "age_to":
+      if ((ageFromOption.value)&&(ageToOption.value)) {
+        filterCriterionAge = (item) => (item.dob.age >= ageFromOption.value)&&(item.dob.age <= ageToOption.value);
+      };
+      break;
+    case "search_name":
+      filterCriterionName = (item) => (item.name.first.toLowerCase().includes(searchNameOption.value.toLowerCase()))||(item.name.last.toLowerCase().includes(searchNameOption.value.toLowerCase()));
```

Comment:
It's not reqirements? but recomendation at this stage. Your function does everything at once. In the future, try to separate logic into simpler parts. This will help you read and maintain your code better.